### DISCUSSION TITLE
part-ii/2015-90/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#280] [Question comments]
Part II 2015 Number Theory: Paper 4, Section I, H 

https://questions.tripos.org/part-ii/2015-90/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Suppose $n$ is not a power of $2$. Then $n$ has an odd factor $q$ not equal to $1$; write $n = mq$. Then we can factor $10^n + 1$ as

$$10^n + 1 = (10^m)^q + 1 = (10^m + 1)(10^q - 10^{q-1} + 10^{q-2} - \dots + 1)$$

So $10^n + 1$ has the non-trivial factor $10^m + 1$ and therefore is not prime.

Now if $n$ is a power of $2$ and $p$ is a prime dividing $10^n + 1$, then $10^n \equiv -1 \pmod{p}$. Therefore $10^n \not\equiv 1 \pmod{p}$ but $10^{2n} \equiv 1 \pmod{p}$, so the order of $10$ $\mod p$ divides $2n$ but does not divide $n$. Since $n$ is a power of $2$, we conclude that the order must be $2n$. Since $10^{p-1} \equiv 1 \pmod{p}$ by Fermat's Little Theorem, we conclude that $2n$ divides $p-1$, so $p \equiv 1 \pmod{2n}$ as required.

The method of Fermat factorisation involves computing $x^2 - N$ for $x = \lceil \sqrt{N} \rceil, \lceil \sqrt{N} \rceil + 1, \lceil \sqrt{N} \rceil + 2, \dots$ until we find that this is equal to $y^2$ for some integer $y$. Then $N = x^2 - y^2 = (x - y)(x + y)$, so we get a non-trivial factorsation of $N$.

Now $\lceil \sqrt{10^4 + 1} \rceil = 101$. We compute $101^2 - N, 102^2 - N, \dots$ and get the following results:

$$
101^2 - N = 200 \\
102^2 - N = 403 \\
103^2 - N = 608 \\
104^2 - N = 815 \\
105^2 - N = 1024 \\
$$

We notice that $1024 = 32^2$, so 

$$N = 105^2 - 32^2 = (105 - 32)(105 + 32) = 73 \cdot 137$$

It is not hard to check (by trial division) that this is a factorisation into primes.


</details>
